src/loader/AMASSLoader.py

```diff
@@ -0,0 +1,320 @@
+        # Pose Specs
+        self._used_sub_dataset_id = self.config.get_string("sub_dataset_id")
+        self._used_subject_id = self.config.get_string("subject_id")
+        self._used_sequence_id = self.config.get_string("sequence_id")
+        self._used_frame_id = self.config.get_string("frame_id")
+        # Get the currently supported mocap datasets by this loader
+        # TODO: file maintaince responsibility
+        taxonomy_file_path = os.path.join(self._data_path, "taxonomy.json")
+        AMASSLoader._get_supported_mocap_datasets(
+            taxonomy_file_path, self._data_path)
+
+    @staticmethod
+    def _get_pose_coefficients(used_sub_dataset_id, used_subject_id,
+                               used_sequence_id, used_frame_id):
+        """
+        Get the pose parameters used to generate mesh object represents the pose
+        :param used_sub_dataset_id: Identifier for the sub dataset, the dataset which the human pose object should be extracted from
+        :param used_subject_id: Type of motion from which the pose should be extracted, this is dataset dependent parameter
+        :param used_frame_id: Frame id in a selected motion sequence.
+        :return: body pose and shape parameters
+        """
+        # check if the sub_dataset is supported
+        if used_sub_dataset_id in AMASSLoader.supported_mocap_datasets:
+            # get  path from dictionsary
+            sub_dataset_path = AMASSLoader.supported_mocap_datasets[used_sub_dataset_id]
+            # concatenate path to specific
+            subject_path = os.path.join(sub_dataset_path, used_subject_id)
+            sequence_path = os.path.join(subject_path,
+                                         "{:02d}".format(int(used_subject_id)) + "_" + "{:02d}".format(
+                                             int(used_sequence_id)) + "_poses.npz")
+            if os.path.exists(sequence_path):
+                # load AMASS dataset sequence file which contains the coefficients for the whole motion sequence
+                sequence_body_data = np.load(sequence_path)
+                # get the number of supported frames
+                no_of_frames_per_sequence = sequence_body_data['poses'].shape[0]
+                if used_frame_id is None or "":
+                    frame_id = random.randint(
+                        0, no_of_frames_per_sequence)  # pick a random id
+                else:
+                    frame_id = int(used_frame_id)
+                # Extract Body Model coefficients
+                if frame_id in range(0, no_of_frames_per_sequence):
+                    # use GPU to accelerate mesh calculations
+                    comp_device = torch.device(
+                        "cuda" if torch.cuda.is_available() else "cpu")
+                    # controls the global root orientation
+                    # root_orient = torch.Tensor(sequence_body_data['poses'][frame_id:frame_id + 1, :3]).to(comp_device)
```

Comment:
Do we need this? Else please remove code which is not longer necessary.

Reply:
removed now